packages/core/src/tracing/instrumentation/cloud/aws/sqs.js

```diff
@@ -252,10 +255,23 @@ function instrumentedReceiveMessage(ctx, originalReceiveMessage, originalArgs) {
       queue: originalArgs[0].QueueUrl || ''
     };
 
+    /**
+     * The MessageAttributeNames attribute is an option that you tell which message attributes you want to see.
+     * As we use message attributes to store Instana headers, if the customer does not set this attribute to All,
+     * we cannot see the Instana headers, so we need to explicitly add them.
+     */
+    const receveingParams = originalArgs[0];
+
+    if (!receveingParams.MessageAttributeNames) {
+      receveingParams.MessageAttributeNames = [];
+    }
+
+    receveingParams.MessageAttributeNames.push('X_INSTANA_L', 'X_INSTANA_S', 'X_INSTANA_T');
```

Comment:
Just checked the docs on that:
> When using ReceiveMessage, you can send a list of attribute names to receive, or you can return all of the attributes by specifying All or .* in your request. You can also use all message attributes starting with a prefix, for example bar.*.

Three questions: 
1) What is the default behaviour if `MessageAttributeNames` is not set at all? Does it default to `All` (receive all attributes by default) or `[]` (receive no attributes by default). If it is the former, we should not add our `X-INSTANA-...` attributes because that would drastically change the behaviour then: Instead of receiving all attributes, only our trace correlation headers would be received. If the default behaviour is the latter (do not receive any attributes), then setting that parameter and adding our attribute names is mandatory and correct, of course.

2) Would we want to omit adding the `X_INSTANA_...` values if we see that `All` or `.*` is already set? (Not sure if it makes a difference to add our headers explicitly, probably not?)

3) If I read the docs correctly, when we add our attribute names, we could also just add one header name prefix, `X_INSTANA.*` instead of explicitly adding all three individually. WDYT?

Reply:
1. As far as I can see, it's empty array, at least is the default in sqs-consumer. So I understood this addition as an improvement of the instrumentation. When instrumenting SQS, I added `['All']` in the test app, which  may be the case we missed this.

2. I will add this logic. I did not want to write much, which is why I didn't add this check, but it makes sense.

3. I did not know that, and it makes total sense to me, thanks

Reply:
Just for history purposes:
 * aws-sdk will not display any message attributes if not explicitly defined in the MessageAttributeNames array.